# Q-9. 

Given the score of students in multiple exams as:
![data](data_Q9.png)
Test the hypothesis that the mean scores of all the students are the same. If not, name the
student with the highest score.


In [1]:
import pandas as pd
import numpy as np
import scipy.stats as stats

## Loading data

In [3]:
df = pd.DataFrame({'Name':['Karan', 'Deepa', 'Karthik', 'Chandan', 'Jeevan'],
                  'Exam_1':[85, 70, 90, 75, 95],
                  'Exam_2':[90, 80, 85, 70, 92],
                  'Final_Exam':[92, 85, 88, 75, 96]})
df

Name  Exam_1  Exam_2  Final_Exam
0    Karan      85      90          92
1    Deepa      70      80          85
2  Karthik      90      85          88
3  Chandan      75      70          75
4   Jeevan      95      92          96

Since we must check the mean scores (numeric variable) of all students (categoric variable), we will perform a **One-way ANOVA** test on the given dataset. The null hypothesis of the oneway annova is that all the groups (students in out case) share the same mean and there is no statistical difference between the means of the different groups. (i.e. All the students have the same mean score)

In [5]:
# Performing one-way ANOVA on the scores
f_value, p_value = stats.f_oneway(df.Exam_1, df.Exam_2, df.Final_Exam)

print("One-way ANOVA:")
print("F-value:", f_value)
print("p-value:", p_value)

One-way ANOVA:
F-value: 0.3236947791164659
p-value: 0.7295954253321559


Since the p-value (0.7295954253321559) is greater than the common significance level of 0.05, we do not have sufficient evidence to reject the null hypothesis. This suggests that there is **no significant difference in the mean scores** between the groups (i.e., the mean scores of all the students are considered to be the same).